src/datasets/summarizations/trend-partial.summarization.ts

```diff
@@ -0,0 +1,103 @@
+import * as math from 'mathjs';
+import { Summary } from './types';
+import { TimeSeriesPoint } from '../metas/types';
+import { cacheSummaries } from './utils/commons';
+import {
+  PointMembershipFunction,
+  MembershipFunction,
+  trapmf,
+  trapmfL,
+  trapmfR,
+} from './libs/protoform';
+import {
+  exponentialMovingAverage,
+  normalizedUniformPartiallyLinearEpsApprox,
+  TimeSeriesPartialTrend,
+} from './libs/trend';
+import { formatX, formatY } from '../../utils/formatters';
+
+export function queryFactory(points: TimeSeriesPoint[]) {
+  return cacheSummaries(() => {
+
+    const smoothedPoints = exponentialMovingAverage(points);
+    const partialTrends = normalizedUniformPartiallyLinearEpsApprox(smoothedPoints, 0.01);

```

Comment:
I realize this is out of scope of this PR, but please name functions as actions (here and elsewhere) i.e normalizeUniformPartialLinearEpsApprox